components/board/fake/board.go

```diff
@@ -289,14 +289,18 @@ func (a *Analog) reset(pin string) {
 	a.Mu.Unlock()
 }
 
-func (a *Analog) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
+func (a *Analog) Read(ctx context.Context, extra map[string]interface{}) (int, board.AnalogRange, error) {
 	a.Mu.RLock()
 	defer a.Mu.RUnlock()
 	if a.pin != analogTestPin {
 		a.Value = a.fakeValue
-		a.fakeValue++
+		if a.fakeValue == 1000 {
+			a.fakeValue = 0
+		} else {
+			a.fakeValue++
```

Comment:
:+1: I really like this new behavior!

(if you want to be fancy, it's just 
```go
a.fakeValue++
a.fakeValue %= 1001
```
with no need for an if statement.)